### PR TITLE
feat: skip clone process for local pipelines

### PR DESCRIPTION
**Pull request by jbrockopp**
Part of the effort for https://github.com/go-vela/community/issues/54

NOTE: The LOC changed in this PR is a bit misleading due to the tests.

## Part 1

Adding a `local` field to the compiler:

https://github.com/go-vela/compiler/blob/3c8cfcb9de429678d357704f850f3a79bf5f6998/compiler/native/native.go#L37

And allowing that field to be set via the `WithLocal()` function:

https://github.com/go-vela/compiler/blob/3c8cfcb9de429678d357704f850f3a79bf5f6998/compiler/native/native.go#L119-L124

## Part 2

Updating the compiler to skip injecting the clone process if the `local` field is set:

https://github.com/go-vela/compiler/blob/3c8cfcb9de429678d357704f850f3a79bf5f6998/compiler/native/clone.go#L21-L27

https://github.com/go-vela/compiler/blob/3c8cfcb9de429678d357704f850f3a79bf5f6998/compiler/native/clone.go#L57-L63

## Part 3

Updating the compiler to set "defaults" for the repo org and name as well as the build number:

https://github.com/go-vela/compiler/blob/3c8cfcb9de429678d357704f850f3a79bf5f6998/compiler/native/transform.go#L34-L49

https://github.com/go-vela/compiler/blob/3c8cfcb9de429678d357704f850f3a79bf5f6998/compiler/native/transform.go#L104-L119

### CODE REVIEW

File: compiler/native/transform_test.go [hunk 1]
```diff
@@ -99,84 +99,158 @@ func TestNative_TransformStages(t *testing.T) {
...
-				Name: "install deps",
-				Steps: pipeline.ContainerSlice{
+	// setup tests
+	tests := []struct {
+		failure  bool
+		local    bool
+		pipeline *yaml.Build
+		want     *pipeline.Build
+	}{
+		{
```
[the remark this answers is not in the document]
Comment (jbrockopp) [reply compiler/native/transform_test.go:109]: Ignoring this 👍

File: compiler/native/transform_test.go [hunk 2]
```diff
@@ -99,84 +99,158 @@ func TestNative_TransformStages(t *testing.T) {
...
-		Secrets: pipeline.SecretSlice{
-			&pipeline.Secret{
-				Name: "foobar",
-				Origin: &pipeline.Container{
-					ID:     "secret___0_vault",
-					Name:   "vault",
-					Image:  "vault:latest",
-					Pull:   "always",
-					Number: 1,
+		{
```
[the remark this answers is not in the document]
Comment (jbrockopp) [reply compiler/native/transform_test.go:164]: Ignoring this 👍

File: compiler/native/transform_test.go [hunk 3]
```diff
@@ -252,66 +326,135 @@ func TestNative_TransformSteps(t *testing.T) {
...
-				Number:      1,
-				Pull:        "always",
+	// setup tests
+	tests := []struct {
+		failure  bool
+		local    bool
+		pipeline *yaml.Build
+		want     *pipeline.Build
+	}{
+		{
```
[the remark this answers is not in the document]
Comment (jbrockopp) [reply compiler/native/transform_test.go:336]: Ignoring this 👍

File: compiler/native/transform_test.go [hunk 4]
```diff
@@ -252,66 +326,135 @@ func TestNative_TransformSteps(t *testing.T) {
...
-		Secrets: pipeline.SecretSlice{
-			&pipeline.Secret{
-				Name: "foobar",
-				Origin: &pipeline.Container{
-					ID:     "secret___0_vault",
-					Name:   "vault",
-					Image:  "vault:latest",
-					Pull:   "always",
-					Number: 1,
+		{
```
[the remark this answers is not in the document]
Comment (jbrockopp) [reply compiler/native/transform_test.go:386]: Ignoring this 👍

File: compiler/native/native.go [hunk 5]
```diff
@@ -42,7 +43,7 @@ type client struct {
 // New returns a Pipeline implementation that integrates with the supported registries.
 func New(ctx *cli.Context) (*client, error) {
```
[the remark this answers is not in the document]
Comment (jbrockopp) [reply compiler/native/native.go:44]: Ignoring this 👍

**Review by wass3r — APPROVED** [review]

File: compiler/native/transform.go [hunk 6]
```diff
@@ -38,6 +38,16 @@ func (c *client) TransformStages(r *pipeline.RuleData, p *yaml.Build) (*pipeline
 	name := c.repo.GetName()
 	number := c.build.GetNumber()
 
+	// check if the compiler is setup for a local pipeline
+	if c.local {
+		// set a default for the org
+		org = "localOrg"
```
Comment (davidvader) [inline compiler/native/transform.go:44]: these could be better suited as constants, but not a big deal

**Review by davidvader — APPROVED** [review]

File: compiler/native/transform.go (hunk 6 above)
Comment (jbrockopp) [reply compiler/native/transform.go:44]: I'll look to make that change in a future iteration 👍

**Pull request merged**